src/DocumentDB.ChangeFeedProcessor/Bootstrapping/PartitionSynchronizer.cs

```diff
@@ -114,6 +118,34 @@ private async Task<List<PartitionKeyRange>> EnumPartitionKeyRangesAsync()
             return partitionKeyRanges;
         }
 
+        private async Task<bool> HasCollectionProvisionedThroughput()
+        {
+            if (this.hasCollectionProvisionedThroughput.HasValue)
+            {
+                return this.hasCollectionProvisionedThroughput.Value;
+            }
+
+            IFeedResponse<Offer> response = null;
+            var offers = new List<Offer>();
+            do
+            {
+                var feedOptions = new FeedOptions
+                {
+                    MaxItemCount = this.maxBatchSize,
+                    RequestContinuation = response?.ResponseContinuation,
+                };
+                response = await this.documentClient.ReadOffersFeedAsync(feedOptions).ConfigureAwait(false);
+                foreach (Offer offer in response)
+                {
+                    offers.Add(offer);
+                }
+            }
+            while (!string.IsNullOrEmpty(response.ResponseContinuation));
+
+            this.hasCollectionProvisionedThroughput = offers.Any(x => x.ResourceId == this.collectionSelfLink);
```

Comment:
it should be x.ResourceLink

Reply:
Update your tests to reflect this change

Reply:
Thanks  @ramarag ! I addressed it in https://github.com/Azure/azure-documentdb-changefeedprocessor-dotnet/pull/121